streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -457,15 +457,9 @@ public long restore(final Map<TaskId, Task> tasks) {
                 //       small batches; this can be optimized in the future, e.g. wait longer for larger batches.
                 final TaskId taskId = changelogs.get(partition).stateManager.taskId();
                 try {
+                    final Task task = tasks.get(taskId);
                     final ChangelogMetadata changelogMetadata = changelogs.get(partition);
-                    final int restored = restoreChangelog(changelogMetadata);
-                    if (restored > 0 || changelogMetadata.state().equals(ChangelogState.COMPLETED)) {
-                        final Task task = tasks.get(taskId);
-                        if (task != null) {
```

Comment:
Why do we remove this `null` check?

Reply:
I pondered on the code and I think it should not be `null` ever? Please correct me if I'm wrong.

Reply:
@mjsax LMK what do you think? I may lack some background here.

Reply:
Was just wondering. It seemed to be unrelated to this PR, and so I just assumed there must be a reason for having the `null` check.